app/src/main/java/com/github/palFinderTeam/palfinder/meetups/activities/MapListViewModel.kt

```diff
@@ -52,31 +54,31 @@ class MapListViewModel @Inject constructor(
      * set the map to which utils functions will be applied
      * @param map: GoogleMap
      */
-    fun setMap(map : GoogleMap){
+    fun setGmap(map : GoogleMap){
         this.map = map
-        mapReady = true
-    }
-    fun getMap(): GoogleMap {
-        return map
+        this.mapReady = true
+        setPositionAndZoom(startingCameraPosition, startingZoom)
     }
 
-    fun update(location: LatLng?){
+    fun update(){
         if(false){//getZoom() < 7f){
             //TODO get only the joined meetup
 
         } else{
-            val earthRadius = 6371000.0
+            val earthCircumference = 40000.0
             // at zoom 0, the map is of size 256x256 pixels and for every zoom, the number of pixel is multiplied by 2
-            val radiusAtZoom0 = earthRadius/256
+            val radiusAtZoom0 = earthCircumference/256
             val radius = radiusAtZoom0/2.0.pow(getZoom().toDouble())
-
-            /*if(meetUpRepository.getAllMeetUps().asLiveData().value != null) {
-                meetUps = meetUpRepository.getAllMeetUps().asLiveData().value!!
-            }else meetUps = emptyList()*/
-            /*listOfMeetUpResponse = meetUpRepository.getMeetUpsAroundLocation(Location(location!!.longitude, location!!.latitude),
-                earthRadius/1000.0).asLiveData()*/
-            listOfMeetUpResponse = meetUpRepository.getAllMeetUpsResponse().asLiveData()
-            refresh()
+            viewModelScope.launch {
+                meetUpRepository.getMeetUpsAroundLocation(
+                    Location(
```

Comment:
We can change this in the future with the new static conversion function that I worked on in `Location`

Reply:
yes, I'm gonna wait a bit for your PR to fix this before merging